salt/cloud/clouds/saltify.py

```diff
@@ -190,3 +317,123 @@ def _verify(vm_):
             return False
+
+
+def destroy(name, call=None):
+    ''' Destroy a node.
+
+    .. versionadded:: xxx
+
+    CLI Example:
+    .. code-block:: bash
+
+        salt-cloud --destroy mymachine
+    '''
+    if call == 'function':
+        raise SaltCloudSystemExit(
+            'The destroy action must be called with -d, --destroy, '
+            '-a, or --action.'
+        )
+
+    opts = __opts__
+
+    __utils__['cloud.fire_event'](
+        'event',
+        'destroying instance',
+        'salt/cloud/{0}/destroying'.format(name),
+        args={'name': name},
+        sock_dir=opts['sock_dir'],
+        transport=opts['transport']
+    )
+
+    local = salt.netapi.NetapiClient(opts)
+    cmd = {'client': 'local',
+           'tgt': name,
+           'fun': 'grains.get',
+           'arg': ['salt-cloud'],
+           }
+    cmd.update(_get_connection_info())
+    vm_ = cmd['vm']
+    my_info = local.run(cmd)
+    try:
+        vm_.update(my_info[name])  # get profile name to get config value
+    except (IndexError, TypeError):
+        pass
+    if config.get_cloud_config_value(
+           'remove_config_on_destroy', vm_, opts, default=True
+            ):
+        cmd.update({'fun': 'service.disable', 'arg': ['salt-minion']})
+        ret = local.run(cmd)  # prevent generating new keys on restart
+        if ret and ret[name]:
+            log.info('disabled salt-minion service on %s', name)
+        cmd.update({'fun': 'config.get', 'arg': ['conf_file']})
+        ret = local.run(cmd)
+        if ret and ret[name]:
+            confile = ret[name]
+            cmd.update({'fun': 'file.remove', 'arg': [confile]})
+            ret = local.run(cmd)
+            if ret and ret[name]:
+                log.info('removed minion %s configuration file %s',
+                         name, confile)
+        cmd.update({'fun': 'config.get', 'arg': ['pki_dir']})
+        ret = local.run(cmd)
+        if ret and ret[name]:
+            pki_dir = ret[name]
+            cmd.update({'fun': 'file.remove', 'arg': [pki_dir]})
+            ret = local.run(cmd)
+            if ret and ret[name]:
+                log.info(
+                    'removed minion %s key files in %s',
+                    name,
+                    pki_dir)
+
+    if config.get_cloud_config_value(
+        'shutdown_on_destroy', vm_, opts, default=False
+        ):
+        cmd.update({'fun': 'system.shutdown', 'arg': ''})
+        ret = local.run(cmd)
+        if ret and ret[name]:
+            log.info('system.shutdown for minion %s successful', name)
+
+    __utils__['cloud.fire_event'](
+        'event',
+        'destroyed instance',
+        'salt/cloud/{0}/destroyed'.format(name),
+        args={'name': name},
+        sock_dir=opts['sock_dir'],
+        transport=opts['transport']
+    )
+
+    return {'Destroyed': '{0} was destroyed.'.format(name)}
+
+
+def reboot(name, call=None):
+    '''
+    Reboot a saltify minion.
+
+    .. versionadded:: xxx
```

Comment:
All of the versionadded tags that you're adding with this PR should be `Oxygen`. Can you update the remaining ones as well, instead of me commenting on them all? :)